frontends/ast/genrtlil.cc

```diff
@@ -849,6 +849,8 @@ void AstNode::detectSignWidthWorker(int &width_hint, bool &sign_hint, bool *foun
 				while (id_ast->simplify(true, false, false, 1, -1, false, true)) { }
 			if (id_ast->children[0]->type == AST_CONSTANT)
 				this_width = id_ast->children[0]->bits.size();
+			else if (id_ast->children.size() == 1 && id_ast->children[0]->type == AST_IDENTIFIER && id_ast->children[0]->id2ast)
+				this_width = id_ast->children[0]->id2ast->range_left - id_ast->children[0]->id2ast->range_right + 1;
```

Comment:
It seems like is a partial step in the direction of supporting indirecting structs through localparams, but it only resolves width detection and not the rest of simplification.

If we'd like to finish supporting this case in a separate PR, let's drop this (partial) change for now.

Once we get around to implementing it, I do have a few comments. First, it seems like `id_ast->type` should be restricted to parameters or localparams in this branch. Second, it would arguably be necessary to check if the target is a struct, etc. Finally, it might be necessary to look up the target iteratively, to support, e.g.,:
```SystemVerilog
localparam foo_t P = ...;
localparam A = P.w;
localparam B = A.x.y;
localparam C = B.z;
```

Reply:
Ok, lets finish supporting whole sub-structs in another PR.
I've dropped support for this in this PR and I will open another one for this.